marconi-chain-index/src/Marconi/ChainIndex/Experimental/Api/HttpServer.hs

```diff
@@ -127,6 +148,81 @@ getEpochNonceHandler
       (Either (JsonRpcErr String) (Core.Result EpochState.NonceByEpochNoQuery))
 getEpochNonceHandler = queryHttpReaderHandler (configQueryables . queryableEpochState)
 
+getMintingPolicyHashTxHandler
+  :: MintTokenEvent.QueryByAssetId MintTokenEvent.MintTokenBlockEvents
+  -> ReaderHandler
+      HttpServerConfig
+      ( Either
+          (JsonRpcErr String)
+          ( Core.Result
+              ( Core.WithStability
+                  (MintTokenEvent.QueryByAssetId MintTokenEvent.MintTokenBlockEvents)
+              )
+          )
+      )
+getMintingPolicyHashTxHandler =
+  queryHttpReaderHandler
+    (configQueryables . queryableMintToken)
+    . Core.WithStability
+
+getBurnTokenEventsHandler
+  :: GetBurnTokenEventsParams
+  -> ReaderHandler HttpServerConfig (Either (JsonRpcErr String) GetBurnTokenEventsResult)
+getBurnTokenEventsHandler = dimapHandler mapQuery mapResults getMintingPolicyHashTxHandler
+  where
+    mapQuery
+      :: GetBurnTokenEventsParams
+      -> MintTokenEvent.QueryByAssetId MintTokenEvent.MintTokenBlockEvents
+    mapQuery (GetBurnTokenEventsParams policyId assetName beforeSlotNo afterTx) =
+      MintTokenEvent.QueryByAssetId
+        policyId
+        assetName
+        (Just MintTokenEvent.BurnEventType)
+        beforeSlotNo
+        afterTx
+
+    -- Map our internal events to our 'getBurnTokenEventsHandler' response object
+    mapResults
+      :: [ Core.Stability
+            ( Core.Timed
+                (Core.Point MintTokenEvent.MintTokenBlockEvents)
+                MintTokenEvent.MintTokenBlockEvents
+            )
+         ]
+      -> GetBurnTokenEventsResult
+    mapResults events = GetBurnTokenEventsResult $ mapResult =<< events
+
+    {- Map a block's worth of internal events to the components of our 'getBurnTokenEventsHandler'
+      response object -}
+    mapResult
+      :: Core.Stability (Core.Timed C.ChainPoint MintTokenEvent.MintTokenBlockEvents)
+      -> [BurnTokenEventResult]
+    mapResult res =
+      let stable = Core.isStable res
+          timed = extract res
+       in mapEvent
+            (timed ^. Core.point)
+            stable
+            <$> NonEmpty.toList (timed ^. Core.event . mintTokenEvents)
+
+    {- Map an internal event to a single component of our 'getBurnTokenEventsHandler' response
+      object -}
+    mapEvent :: C.ChainPoint -> Bool -> MintTokenEvent.MintTokenEvent -> BurnTokenEventResult
+    mapEvent chainPoint isStable event =
+      let (slotNo, headerHash) = case chainPoint of
+            (C.ChainPoint slot hash) -> (slot, hash)
+            C.ChainPointAtGenesis -> (0, "genesis")
```

Comment:
I'm not sure about this bit in particular, is it correct?

Reply:
There shouldn't be any minting/burning at genesis. Although, for completeness, I would probably prefer to use `Maybe SlotNo` and `Maybe BlockHeaderHash` rather than forging values.